src/components/LabelFilterInput.tsx

```diff
@@ -0,0 +1,88 @@
+import { GrafanaTheme, SelectableValue } from '@grafana/data';
+import { CascaderOption, MultiSelect, ButtonCascader, useStyles } from '@grafana/ui';
+import React, { useMemo } from 'react';
+import { Check } from 'types';
+import { css } from 'emotion';
+
+interface Props {
+  checks: Check[];
+  labelFilters: string[];
+  className?: string;
+  onChange: (labels: string[]) => void;
+}
+
+interface AggregateLabels {
+  [key: string]: string[];
+}
+
+const getStyles = (theme: GrafanaTheme) => ({
+  prefix: css`
+    margin-left: -${theme.spacing.sm};
+    z-index: 1000;
+    height: 100%;
+  `,
+  cascader: css`
+    border-top-right-radius: 0px;
+    border-bottom-right-radius: 0px;
+  `,
+});
+
+export const LabelFilterInput = ({ checks, labelFilters, onChange, className }: Props) => {
+  const styles = useStyles(getStyles);
+  const aggregatedLabels = useMemo(
+    () =>
+      checks.reduce<AggregateLabels>((acc, check) => {
+        check.labels?.forEach(({ name, value }) => {
+          if (acc[name]) {
+            acc[name].push(value);
+          }
+          acc[name] = [value];
```

Comment:
it seems like this line should be inside an `else` branch

Reply:
Yes! Good catch!